src/PhpSpreadsheet/Style/NumberFormat.php

```diff
@@ -502,6 +502,10 @@ private static function formatAsDate(&$value, &$format): void
         $format = preg_replace_callback('/"(.*)"/U', ['self', 'escapeQuotesCallback'], $format);
 
         $dateObj = Date::excelToDateTimeObject($value);
+        // If the colon preceding minute had been quoted, as happens in
+        // Excel 2003 XML formats, m will not have been changed to i above.
+        // Change it now.
```

Comment:
I believe that Google Docs Spreadsheets also quotes these colons, so it will resolve the issue for them as well